### PR TITLE
ckb2021 on rust2021

### PR DESCRIPTION
**Pull request by zhangsoledad**
### What is changed and how it works?

Upgrate rust toolchain to 1.56.1, and switch to 2021 edition.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by zhangsoledad** [comment]
#benchmark

File: .cargo/config copy [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[cargo-new]
```
Comment (yangby-cryptape) [inline .cargo/config copy:1]: Duplicate?

**Review by yangby-cryptape — APPROVED** [review]

File: .cargo/config copy (hunk 1 above)
Comment (zhangsoledad) [reply .cargo/config copy]: https://github.com/nervosnetwork/ckb/compare/200e072e494a6eeac2640221bb8525de18197e1b..7311da6bcea68072eac102595050610bae354010

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/ckb-docker-builder/pull/7

**Review by quake — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors r=yangby-cryptape,quake

**Pull request merged**